src/process/waitpid_future.cr

```diff
@@ -0,0 +1,24 @@
+# :nodoc:
+class Process::WaitpidFuture
+  def initialize pid
+    @chan = Event::SignalChildHandler.instance.waitpid pid
+    @has_value = false
+    @value = nil
+  end
+
+  def value
+    receive
+
+    if value = @value
+      return value
+    else
+      raise "waitpid channel closed"
```

Comment:
I'm not sure what that should be.  It only occurs when in the child after forking while another fiber is waiting for different process to exit (before forking).

I could return nil but then the type changes to an optional.  
Or raise a custom exception?

Suggestions?